### PR TITLE
Expose admin server for Account Server and enable gRPC health checking service

### PR DESCRIPTION
**Pull request by yashykt**

### CODE REVIEW

File: cpp/utility.cc [hunk 1]
```diff
@@ -87,4 +88,14 @@ std::shared_ptr<grpc::ServerCredentials> GetServerCredentials(
   assert(0);
 }
 
+std::unique_ptr<grpc::Server> StartAdminServer(const std::string& port) {
+  std::string server_address = "localhost" + port;
+  grpc::EnableDefaultHealthCheckService(true);
```
Comment (lidizheng) [inline cpp/utility.cc:93]: Is C++'s health checking service shared within the process or separated by server objects?

**Review by lidizheng — APPROVED** [review]
Thanks. Nice to learn we have an utility file to put common code.

File: cpp/utility.cc (hunk 1 above)
Comment (sanjaypujare) [reply cpp/utility.cc:93]: I don't see the `port` being passed to `grpc::EnableDefaultHealthCheckService()` so how does this code ensure the health checking is started on `port` (which is the value of `admin_port`)?

Comment (yashykt) [reply cpp/utility.cc:93]: The listening port(s) for a server is different from the list of services registered with it. What `grpc::EnableDefaultHealthCheckService()` does is, it registers a health check service by default to servers, which allows the health check service to be exposed on the ports that it serves on.

Comment (sanjaypujare) [reply cpp/utility.cc:93]: Okay it seems to be a global setting that affects all the servers created after this call. In that case the AdminServer will have the health checking enabled. LG

File: cpp/utility.cc [hunk 2]
```diff
@@ -87,4 +88,14 @@ std::shared_ptr<grpc::ServerCredentials> GetServerCredentials(
...
+  grpc::ServerBuilder builder;
+  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+  grpc::AddAdminServices(&builder);
+  return builder.BuildAndStart();
+  std::cout << "Admin Server listening on " << server_address << std::endl;
```
Comment (sanjaypujare) [inline cpp/utility.cc:98]: The Java example emits the message "Admin & health server started, listening on...". How about adding " & health" after "Admin" in this message?

File: cpp/utility.cc (hunk 1 above)
Comment (yashykt) [reply cpp/utility.cc:93]: > Is C++'s health checking service shared within the process or separated by server objects?

Each server uses a different health check service but the implementation itself can be shared. This is the implementation for the default health check service  - https://github.com/grpc/grpc/blob/master/src/cpp/server/health/default_health_check_service.cc

File: cpp/utility.cc (hunk 2 above)
Comment (yashykt) [reply cpp/utility.cc]: Done

**Review by sanjaypujare — APPROVED** [review]
LGTM. Made a suggestion about a small log message change

**Pull request merged**